src/Microsoft.Data.Analysis/TextFieldParser.cs

```diff
@@ -0,0 +1,642 @@
+            builder.Append(_buffer, _position, _charsRead - _position);
+            builder.Append(_reader.ReadToEnd());
+            FinishReading();
+            return builder.ToString();
+        }
+
+        public void Close()
+        {
+            CloseReader();
+        }
+
+        public void Dispose()
+        {
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                if (!_disposed)
+                {
+                    Close();
+                }
+                _disposed = true;
+            }
+        }
+
+        private void ValidateFieldTypeEnumValue(FieldType value, string paramName)
+        {
+            if (value < FieldType.Delimited || value > FieldType.FixedWidth)
+            {
+                throw new InvalidEnumArgumentException(paramName, (int)value, typeof(FieldType));
+            }
+        }
+
+        ~TextFieldParser()
+        {
+            Dispose(disposing: false);
+        }
+
+        private void CloseReader()
+        {
+            FinishReading();
+            if (_reader != null)
+            {
+                if (!_leaveOpen)
+                {
+                    _reader.Close();
+                }
+                _reader = null;
+            }
+        }
+
+        private void FinishReading()
+        {
+            _lineNumber = -1L;
+            _endOfData = true;
+            _buffer = null;
+        }
+
+        private void InitializeFromPath(string path, Encoding defaultEncoding, bool detectEncoding)
+        {
+            if (path == null)
+            {
+                throw new ArgumentNullException(nameof(path));
+            }
+            if (defaultEncoding == null)
+            {
+                throw new ArgumentNullException(nameof(defaultEncoding));
+            }
+            string fullPath = ValidatePath(path);
+            FileStream fileStreamTemp = new FileStream(fullPath, (FileMode.Open), (FileAccess.Read), (FileShare.ReadWrite));
+            _reader = new StreamReader(fileStreamTemp, defaultEncoding, detectEncoding);
+            ReadToBuffer();
+        }
+
+        private void InitializeFromStream(Stream stream, Encoding defaultEncoding, bool detectEncoding)
+        {
+            if (stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+            if (!stream.CanRead)
+            {
+                throw new ArgumentException("stream can't read");
+            }
+            if (defaultEncoding == null)
+            {
+                throw new ArgumentNullException(nameof(defaultEncoding));
+            }
+            _reader = new StreamReader(stream, defaultEncoding, detectEncoding);
+            ReadToBuffer();
+        }
+
+        private string ValidatePath(string path)
+        {
+            if (!File.Exists(path))
```

Comment:
Just to point out: this doesn't guarantee that at the time of use of the file it actually exists.
The file could be deleted between this check and the actual use. So there's a race condition.

But I would leave it as is, because it's a best effort solution which is perfectly fine.